### PR TITLE
WCOSS2 run support and script/config updates

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
**Description**

This PR includes script and run support updates for running the `develop` branch on WCOSS2. This PR does not include resource updates for running on WCOSS2 and does not complete the port to WCOSS2. Additional WCOSS2 `develop` port PRs will follow this PR.

Changes in this PR mostly address the following:

1. WCOSS2 port changes requested by NCO (e.g. `NWPROD`/`NWROOT` -> `PACKAGEROOT`) in the move of operations to WCOSS2 in the past year.
2. Adding WCOSS2 support in the way of creating `WCOSS2.env`, a `wcoss2.yaml` hosts file for rocoto, and config updates.
3. Some cleanup that was also done in the operations GFS port to WCOSS2.

More specifically, the changes in this PR are:

1. Create `WCOSS2.env` under the env folder.
2. Remove references/usage of jlogfile thoughout.
3. Change `NWPROD` to `PACKAGEROOT` throughout.
4. Replace `NWROOT` with `OPSROOT` where appropriate.
5. Env file updates (non-WCOSS2):
  * Update "anal" if block in `HERA.env` and `ORION.env` to add "or" for "analcalc"; needed to match similar setting in `WCOSS2.env` and change in `jobs/JGLOBAL_ATMOS_ANALYSIS_CALC`
6. Update JJOB scripts:
  * Update `jobs/JGLOBAL_ATMOS_ANALYSIS_CALC` to pass "analcalc" to env file instead of "anal"
  * Move config and env file sourcing from rocoto job scripts into JJOBs for gempak
  * Update the default COM* paths to utilize `compath.py` and the new `compath.py` format; includes updating `COMINukmet` and `COMINecmwf` defaults to new `compath.py` format
  * Add `COMINobsproc` to `jobs/JGFS_ATMOS_GEMPAK_NCDC_UPAPGIF`
  * Replace `COMINgfs` with `COMINobsproc` and replace `COMINgfs_m6hrs` with `COMIN_m6hrs` in `jobs/JGLOBAL_ATMOS_EMCSFC_SFC_PREP`
  * Update `ARCHSYND`, `HOMENHC` and `TANK_TROPCY` in `jobs/JGLOBAL_ATMOS_TROPCY_QC_RELOC`, remove `HOMENHCp1`
  * Update default `APRUN` commands for several downstream jobs (`mpirun` -> `mpiexec`)
  * Set `wavempexec` to `wavempexec` instead of `launcher` in `jobs/JGLOBAL_WAVE_INIT`
  * Correct list of configs to source in `jobs/JGLOBAL_WAVE_POST_BNDPNTBLL` to include `wavepostbndpntbll` instead `wavepostbndpnt`
  * Update how `PDY` is invoked (`. PDY` to `. ./PDY`)
  * Update the `RUN` and `RUN2` variables in `jobs/JGFS_ATMOS_GEMPAK_PGRB2_SPEC`
  * Some cleanup to remove unneeded `RUN_ENVIR=nco` checks and some extra whitespaces.
7. Update rocoto job scripts:
  * Move config and env sourcing out of gempak rocoto job scripts (moved to JJOB scripts); also change to invoke JJOB directly and add `exit $status`
  * Update fit2obs and tracker section of `jobs/rocoto/vrfy.sh`
  * Add missing env file sourcing to `jobs/rocoto/wafsgcip.sh`
8. Update `module_base.wcoss2.lua` to use para version of `ncdiag` module (update to use official ops install later)
9. Updates to configs:
  * Whitespace cleanup in `config.anal`
  * Add `export nth_echgres=$nth_echgres_gfs` when CDUMP=gfs in `config.analcalc`
  * Add `COMINsyn` to `config.base.emc.dyn`
  * Update `ROTDIR` in `config.base.nco.static`
  * Change `OUTPUT_FILETYPE_` variables to be "netcdf" on all machines in `config.efcs`
  * Updates to `config.fv3.nco.static`: remove non-WCOSS2 references, add `npe_node_max=128` for WCOSS2, change C384 DELTIM to 200 (ops value), and update variables for C384/C768 to ops values
  * Remove `GEMPAKSH=$HOMEgfs/jobs/JGFS_ATMOS_GEMPAK` line in `config.gempak`; now invoking `$HOMEgfs/jobs/JGFS_ATMOS_GEMPAK` directly in the JJOB
  * Update `COMINsyn` default in `config.prep` and `config.vrfy`
  * Create `config.resources.nco.static`, copy from `operations`; will need updating for GFSv17 opd
  * Tracker updates in `config.vrfy`
10. Updates to scripts:
  * Updates to `RUN` and `RUN2` variables in `scripts/exgdas_atmos_nawips.sh`, `scripts/exgfs_atmos_goes_nawips.sh`, and `scripts/exgfs_atmos_nawips.sh`
  * Replace `COMINgfs` with `COMINobsproc` in `scripts/exgfs_atmos_gempak_gif_ncdc_skew_t.sh`
  * Update `APRUN` in `scripts/exgfs_atmos_gempak_meta.sh` (`mpirun` -> `mpiexec`)
11. Update `sorc/link_workflow.sh` to also copy the appropriate versions of `config.fv3` and `config.resources` in NCO mode.
12. Updates to ush scripts:
  * Remove references to `NWPROD` and replace with appropriate variables or paths.
  * Add echos to `ush/gfs_bufr.sh` to print out the specific forecast hour files for SPAs to check in case of failure. (Added by GDIT and Bo Cui)
  * Delete `ush/global_extrkr.sh` (not used anywhere anymore).
13. Update util source codes to build with `ftn` and include `-g -traceback` `FFLAGS`
14. Updates to setup script files:
  * Replace `nwprod` with `packageroot` in hosts files and `setup_expt.py`.
  * Create `wcoss2.yaml` host file.
  * Update `workflow_tasks.py` to set native xml tag for pbspro.

Issue #419

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

- [x] Clone and build tests on WCOSS2, Hera, and Orion
- [x] Cycled atmos-only tests on WCOSS2, Hera, and Orion (C192C96L127 on all machines and C768C384L127 on WCOSS2)
- [x] Forecast-only coupled tests on Hera and Orion

No downstream/ops-only jobs were invoked in testing, however updates to those job scripts are included in this PR.
  
**Checklist**

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by KateFriedman-NOAA** [comment]
Tests on Hera and Orion have both now completed. Atmos-only cycled and forecast-only coupled tests using `feature/dev-wcoss2` all reproduced controls using `develop`.

File: workflow/hosts/hera.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 base_git: '/scratch1/NCEPDEV/global/glopara/git'
 base_svn: '/scratch1/NCEPDEV/global/glopara/svn'
 dmpdir: '/scratch1/NCEPDEV/global/glopara/dump'
-nwprod: '/scratch1/NCEPDEV/global/glopara/nwpara'
-comroot: '/scratch1/NCEPDEV/global/glopara/com'
+packageroot: '/scratch1/NCEPDEV/global/glopara/nwpara'
+comroot: '/scratch1/NCEPDEV/rstprod/prod/com'
```
Comment (WalterKolczynski-NOAA) [inline workflow/hosts/hera.yaml]: Why is comroot changed on hera?

File: util/sorc/webtitle.fd/makefile [hunk 2]
```diff
@@ -11,12 +11,12 @@ OBJS=   webtitle.o
 # CMD		Name of the executable
 # PROFLIB	Library needed for profiling
 #
-FC =	ifort 
+FC =	ftn
```
Comment (WalterKolczynski-NOAA) [inline util/sorc/webtitle.fd/makefile]: The util directory is being removed by PR #1009. Any changes to these files need to be made as a PR to the [NOAA-EMC/gfs-util](https://github.com/NOAA-EMC/gfs-utils) repo

File: jobs/JGDAS_ATMOS_GEMPAK [hunk 3]
```diff
@@ -5,6 +5,24 @@ source "$HOMEgfs/ush/preamble.sh"
 ############################################
 # GDAS GEMPAK PRODUCT GENERATION
 ############################################
+echo
+echo "=============== BEGIN GEMPAK ==============="
+echo
+echo "=============== BEGIN TO SOURCE RELEVANT CONFIGS ==============="
```
Comment (WalterKolczynski-NOAA) [inline jobs/JGDAS_ATMOS_GEMPAK]: The preamble announces the script beginning, it isn't needed here.

File: jobs/JGFS_ATMOS_GEMPAK [hunk 4]
```diff
@@ -5,13 +5,31 @@ source "$HOMEgfs/ush/preamble.sh"
 ############################################
 # GFS GEMPAK PRODUCT GENERATION
 ############################################
+echo
+echo "=============== BEGIN GEMPAK ==============="
```
Comment (WalterKolczynski-NOAA) [inline jobs/JGFS_ATMOS_GEMPAK]: Same here. The preamble announces the script start.

File: jobs/JGLOBAL_ATMOS_NCEPPOST [hunk 5]
```diff
@@ -70,7 +70,7 @@ export COMPONENT=${COMPONENT:-atmos}
 ##############################################
 export APRUNP=${APRUN:-$APRUN_NP}
 export RERUN=${RERUN:-NO}
-export HOMECRTM=${HOMECRTM:-${NWROOT:-}/lib/crtm/${crtm_ver:-}}
+export HOMECRTM=${HOMECRTM:-${PACKAGEROOT:-}/lib/crtm/${crtm_ver:-}}
```
Comment (WalterKolczynski-NOAA) [inline jobs/JGLOBAL_ATMOS_NCEPPOST]: With empty default values for PACKAGEROOT and crtm_ver here, this line will produce incorrect paths but not error.

File: parm/config/config.base.emc.dyn [hunk 6]
```diff
@@ -76,9 +77,8 @@ export MODE="@MODE@" # cycled/forecast-only
 export FIXgsi="$HOMEgfs/fix/fix_gsi"
 export HOMEfv3gfs="$HOMEgfs/sorc/fv3gfs.fd"
 export HOMEpost="$HOMEgfs"
-export HOMEobsproc_prep="$BASE_GIT/obsproc/obsproc_prep.v5.5.0_hpc-stack"
-export HOMEobsproc_network="$BASE_GIT/obsproc/obsproc_global.v3.4.2_hpc-stack"
-export HOMEobsproc_global=$HOMEobsproc_network
+#export HOMEobsproc="$BASE_GIT/obsproc/v1.1.0"
```
Comment (WalterKolczynski-NOAA) [inline parm/config/config.base.emc.dyn]: Remove dead code. Why are we not using the v1.1.0?

File: parm/config/config.resources.nco.static [hunk 7]
```diff
@@ -0,0 +1,354 @@
+#!/bin/ksh -x
```
Comment (WalterKolczynski-NOAA) [inline parm/config/config.resources.nco.static]: I know this is copied from ops branch, but we've already converted all the others to bash.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: workflow/hosts/hera.yaml (hunk 1 above)
Comment (KateFriedman-NOAA) [reply workflow/hosts/hera.yaml]: I have reverted it back to the prior setting. I had changed it to the new prod mirror location (/scratch1/NCEPDEV/rstprod/prod/com) so the tracker could get the syndat files but NCO won't add syndat to the mirror until next year (put in request last week).

Instead, I have:
1. Updated the existing syndat folder under /scratch1/NCEPDEV/global/glopara/com on Hera and /work/noaa/global/glopara/com/gfs/prod/syndat on Orion with the current syndat from ops on WCOSS2.
2. Added `cominsyn` paths to the hosts files.
3. Updated the setup scripts and `config.base.emc.dyn` to set `COMINsyn` using `cominsyn` from the hosts files.

File: parm/config/config.base.emc.dyn (hunk 6 above)
Comment (KateFriedman-NOAA) [reply parm/config/config.base.emc.dyn]: I have reverted this back to the prior HOMEobsproc* variables. Need additional updates for the new prepobs modulefile so I will update obsproc/prepobs in a separate PR.

Comment (KateFriedman-NOAA) [reply parm/config/config.base.emc.dyn]: Also, obsproc.v1.1.0 is coming with GFSv16.3, we're at v1.0.2 for GFSv16.2. Waiting for final tag of v1.1.0, may update to v1.0.2 in `develop` first. The prepobs version is v1.0.1 in both GFS versions.

File: jobs/JGLOBAL_ATMOS_NCEPPOST (hunk 5 above)
Comment (KateFriedman-NOAA) [reply jobs/JGLOBAL_ATMOS_NCEPPOST]: Good catch. Should we fix the empty defaults in this PR or your task to remove "NCEP" from the post script names?

Comment (WalterKolczynski-NOAA) [reply jobs/JGLOBAL_ATMOS_NCEPPOST]: Here. Just throw in some question marks if you need to.

File: jobs/JGDAS_ATMOS_GEMPAK (hunk 3 above)
Comment (KateFriedman-NOAA) [reply jobs/JGDAS_ATMOS_GEMPAK]: I removed lines 8 and 9. Should lines 10 and 11 go as well?

File: jobs/JGFS_ATMOS_GEMPAK (hunk 4 above)
Comment (KateFriedman-NOAA) [reply jobs/JGFS_ATMOS_GEMPAK]: Removed.

File: parm/config/config.resources.nco.static (hunk 7 above)
Comment (KateFriedman-NOAA) [reply parm/config/config.resources.nco.static]: Converted to bash. Thanks for catching!

File: jobs/JGDAS_ATMOS_GEMPAK (hunk 3 above)
Comment (WalterKolczynski-NOAA) [reply jobs/JGDAS_ATMOS_GEMPAK]: Probably, if only because other scripts don't do it.

File: parm/config/config.base.emc.dyn (hunk 6 above)
Comment (WalterKolczynski-NOAA) [reply parm/config/config.base.emc.dyn]: Okay, I just saw the nco version of this file using 1.1.0, so I was wondering why the difference.

File: jobs/JGLOBAL_ATMOS_NCEPPOST (hunk 5 above)
Comment (KateFriedman-NOAA) [reply jobs/JGLOBAL_ATMOS_NCEPPOST]: Done!

File: parm/config/config.base.emc.dyn (hunk 6 above)
Comment (KateFriedman-NOAA) [reply parm/config/config.base.emc.dyn]: That was me getting ahead of things to set the incoming ops version in the nco version of that config. I'll redo the nco config obsproc update when I do the dev side change now instead.

File: jobs/JGDAS_ATMOS_GEMPAK (hunk 3 above)
Comment (KateFriedman-NOAA) [reply jobs/JGDAS_ATMOS_GEMPAK]: Done. I also removed the other "BEGIN TO SOURCE" announcement for the env file.

File: jobs/JGFS_ATMOS_GEMPAK (hunk 4 above)
Comment (KateFriedman-NOAA) [reply jobs/JGFS_ATMOS_GEMPAK]: I also removed the other "BEGIN TO SOURCE" announcement for the env file.

File: util/sorc/webtitle.fd/makefile (hunk 2 above)
Comment (KateFriedman-NOAA) [reply util/sorc/webtitle.fd/makefile]: Reverted, Will address in `gfs-util` repo as needed.

**Comment by KateFriedman-NOAA** [comment]
Sync merged with `develop` and resolved conflict in `workflow/hosts/orion.yaml`.

**Comment by KateFriedman-NOAA** [comment]
@WalterKolczynski-NOAA I believe I have addressed your comments. Let me know if not or if you have any others. Thanks!

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**